src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -565,14 +567,14 @@ def test_fw_binaryoperator(fort_writer):
         "subroutine tmp(a,n)\n"
         "  integer, intent(in) :: n\n"
         "  real, intent(out) :: a(n)\n"
-        "    a = sign(1.0,1.0)\n"
+        "    a = {0}(1.0,1.0)\n"
         "end subroutine tmp\n"
-        "end module test")
+        "end module test").format(binary_intrinsic)
     schedule = create_schedule(code, "tmp")
 
     # Generate Fortran from the PSyIR schedule
     result = fort_writer(schedule)
-    assert "a=SIGN(1.0, 1.0)" in result
+    assert "a={0}(1.0, 1.0)".format(binary_intrinsic.upper()) in result
```

Comment:
Again, this isn't valid code for the SUM intrinsic. I know we can't catch this without type information but the test itself should at least be with correct code :-)

Reply:
Created separate test.